data_model/src/lib.rs

```diff
@@ -604,6 +608,88 @@ model! {
         /// `Fixed` value
         Fixed(fixed::Fixed),
     }
+
+    /// Subset of [`Value`] which implements [`Ord`] and [`PartialOrd`].
```

Comment:
I can only assume this is because `permission::Validator` cannot implement `PartialOrd` and `Ord`?

Reply:
> I can only assume this is because `permission::Validator` cannot implement `PartialOrd` and `Ord`?

Exactly


> Why not? since it implements `Eq`

I think, there is no meaningful way to say if one wasm blob is *less* than another.

Reply:
eh, but that now forces you to split the `Value`s into comparable and uncomparable

> I think, there is no meaningful way to tell if one wasm blob is _less_ than another. Although we still can tell if one wasm blob is _the same_ as another

this reasoning is a bit moot because it can be applied to many of `Value` variants. At some point I removed `Ord` from expressions because I was thinking the same: what does it mean that one expression is __less__ then another? I think it was you who brought it back. IMO anything that can implement `Ord` can be ordered

Reply:
> I think it was you who brought it back

Hm, I can't remember it. Although I think that we can possibly compare expression results.

> IMO anything that can implement Ord can be ordered

IMO if it has no meaning to implement `Ord` for a particular struct it shouldn't implement it, because it can lead to a surprising behaviour. If you need that type to implement `Ord` in a particular place you should use *new-type pattern* and implement `Ord` in a particular way you need.

@appetrosyan , what do you think?

Reply:
@Erigara

Reply:
Ok, I did a rollback, check out new version